tests/cpp/test_gpu3.cpp

```diff
@@ -8171,95 +8171,171 @@ TEST_F(NVFuserTest, BroadcastFromNowhere) {
   };
 
   std::unordered_set<IterDomain*> all_ids;
-  auto tv0 = makeSymbolicTensor(1);
+  auto tv0 = makeSymbolicTensor(1); // [I]
   all_ids.insert(tv0->axis(0));
   EXPECT_EQ(all_ids, us(tv0->getLoopDomain()));
   EXPECT_EQ(all_ids, us(tv0->domain()->allIDs()));
   ir_utils::validateDomainEquivalence(
-      tv0->getLoopDomain(), tv0->getLogicalDomain());
+      tv0->getLoopDomain(),
+      tv0->getLogicalDomain(),
+      tv0->domain()->additionalIDs());
+  ir_utils::validateDomainEquivalence(
+      tv0->getLogicalDomain(),
+      tv0->getLoopDomain(),
+      tv0->domain()->additionalIDs());
 
-  tv0->broadcast(0);
+  tv0->broadcast(0); // [b, I]
   EXPECT_TRUE(tv0->axis(0)->isBroadcast());
+  EXPECT_EQ(tv0->axis(0)->extent()->value(), 1);
   EXPECT_EQ(all_ids.count(tv0->axis(0)), 0);
   all_ids.insert(tv0->axis(0));
   EXPECT_EQ(all_ids, us(tv0->getLoopDomain()));
   EXPECT_EQ(all_ids, us(tv0->domain()->allIDs()));
   ir_utils::validateDomainEquivalence(
-      tv0->getLoopDomain(), tv0->getLogicalDomain());
+      tv0->getLoopDomain(),
+      tv0->getLogicalDomain(),
+      tv0->domain()->additionalIDs());
+  ir_utils::validateDomainEquivalence(
+      tv0->getLogicalDomain(),
+      tv0->getLoopDomain(),
+      tv0->domain()->additionalIDs());
 
-  tv0->broadcast(2);
+  tv0->broadcast(2, 2); // [b, I, b]
   EXPECT_TRUE(tv0->axis(2)->isBroadcast());
+  EXPECT_EQ(tv0->axis(2)->extent()->value(), 2);
```

Comment:
Is there a reason this option should affect `extent()` and not `expandedExtent()`?

Reply:
There is no reason. I don't think we have a clear definition on when to use extent and when to use expanded extent. I don't see any reason why expand is not implemented as just using extent, instead of create a new field called "expanded extent".

So far, expanded extent attaches to `ExpandOp`, and if you do `split(b1, 5)`, you will get a broadcasting ID whose extent is 5. For this case, because it is a scheduling activity, I feel that it looks more like the split case.

Reply:
> I don't see any reason why expand is not implemented as just using extent, instead of create a new field called "expanded extent".

Yeah it's a common point of confusion. I don't see any reason we couldn't get rid of `expandedExtent`. It's still useful to have `hasExpandedExtent()` but I guess that could be implemented as `!extent()->sameAs(fusion()->oneVal())`.